source/material_model/visco_plastic.cc

```diff
@@ -248,6 +248,28 @@ namespace aspect
           // Apply strain weakening of the viscous viscosity
           viscosity_pre_yield *= viscous_weakening;
 
+          // For composition spcrust, change from fixed *maximum* viscosity to flow-law viscosity
```

Comment:
Potential rewording: `For composition spcrust` -> `If there is a compositional field named spcrust, change the viscosity from the fixed ....`

Reply:
I've reworded this starting from your suggestion